src/transformations.jl

```diff
@@ -8,63 +8,106 @@ abstract type AbstractDataTransform end
 
 Apply transformation `t` to vector or matrix `x` in place.
 """
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(x, t, x)
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(x, t, x)
+function transform!(t::AbstractDataTransform, x::AbstractMatrix{<:Real})
+    if t.dims == 1
+        transform!(x, t, x)
+    elseif t.dims == 2
+        transform!(x', t, x')
+    end
+    return x
+end
+
+function transform!(t::AbstractDataTransform, x::AbstractVector{<:Real})
+    transform!(t, reshape(x, :, 1))
+    return vec(x)
+end
 
 """
     transform(t::AbstractDataTransform, x)
 
-Return a row-standardized vector or matrix `x` using `t` transformation.
+Return a standardized vector or matrix `x` using `t` transformation.
 """
-transform(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(similar(x), t, x)
-transform(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(similar(x), t, x)
+function transform(t::AbstractDataTransform, x::AbstractMatrix{<:Real})
+    if t.dims == 1
+        return transform!(similar(x), t, x)
+    elseif t.dims == 2
+        return transform!(similar(x)', t, x')'
+    end
+end
+
+function transform(t::AbstractDataTransform, x::AbstractVector{<:Real})
+    xmat = transform(t, reshape(x, :, 1))
```

Comment:
```suggestion
   vec(transform(t, reshape(x, :, 1)))
```
and remove next line.